text/069-coq-roadmap.md

```diff
@@ -0,0 +1,316 @@
+  their own release schedule and versioning scheme, in case they wish to
+  do so.
+
+- Extract the prelude + a minimum set of components that alternative
+  general libraries like MathComp and coq-stdpp need as a basis.
+  Make sure that this reduced core stdlib component can evolve to allow
+  other libraries to use newer features of Coq (like universe polymorphism,
+  SProp and primitive projections).
+  Stop including any other stdlib components as part of the `coq` opam
+  package and encourage maintainers of Coq packages in other distributions
+  to do the same.
+
+Resources:
+
+- Cyril Cohen, Pierre Roux
+- 6 months to 1 year
+
+## Other axes, without sufficient resources
+
+### Kernel, theory
+
+#### Observational type theory
+
+#### Fixpoints
+
+- Global fixpoints: Hugo less convinced of the importance of global fixpoints vs modifying the fix/cofix rules of CIC so that they unfold for named fixpoints on the name rather than the body of the fix
+- Fixpoints able to treat nested inductive types as mutual inductive types (recomputing the recursive structure dynamically), Hugo (PR #17950), a few weeks
+- Guard condition able to detect uniform parameters of inner fixpoints, Hugo (PR #17986), a few weeks
+- Expanded constructors of a branch in a `match` considered smaller for the guard condition (CEP #73)
+
+#### Primitive projections
+
+- Removal of compatibility layer
+
+#### Sections
+
+- Design of a way to refer to the generalized version of a constant/inductive from within the inside of a section
+
+#### Modules
+
+### Surface language
+
+#### AST / interpretation
+
+- Emilio?
+
+#### Bi-dimensional notations
+
+- Emilio Jésus Gallego Arias, ??? (missing buddy)
+
+#### Nametab / libobject
+
+- Emilio Jésus Gallego Arias, ??? (missing buddy)
+
+#### Removing clenv
+
+#### Unifall
+
+#### Unification heuristics
+
+(improving evarconv)
+
+#### Support for async tactics
+
+#### Reactive elaboration
+
+#### Functionalisation
+
+### Libraries and community
+
+#### Promote the Coq Platform
+
+The Coq Platform is already the officially recommended installation method.
+We would like to make it clear to users that they are encouraged to rely on
+the packages that it provides, and that libraries in the Platform should be
+generally considered as a collaborative extended standard library for Coq
+(the historical stdlib being only a part of this).
+
+As part of this effort we should do:
+
+- Editorial work for curation of packages included in the Coq Platform
+  (to bring guarantees on their level of quality).
+
+- Consolidation of documentation, so that it is easy for users to find
+  documentation about the included packages, ideally with a consistent
+  presentation.
+
+- Infrastructure / automation work to stabilize the release process of
+  the Coq Platform and ensure that releases are more consistently
+  delivered according to a predefined schedule.
+
+#### Tooling for building libraries
+
```

Comment:
How about listing improvements to the Ltac* debugger?  I have several significant enhancements already complete.  Only a couple more features to complete, cleanup and doc.  It may be ready to present to the group in a month or less.

Reply:
Feel free to contribute additional items like this one to this list. If items have a single person committed to them, they should appear in the second part, until they get at least a committed reviewer.

Note however that improvements to the debugger (if they are CoqIDE-oriented) risk clashing with some other goals that are likely to get on the roadmap to eventually retire CoqIDE. Adding debugger support to one of the language servers would then become a much higher priority than improving the debugger, IMHO.